indico/modules/events/registration/client/js/components/ParticipantListAccordion.tsx

```diff
@@ -0,0 +1,292 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {HTMLAttributes, ReactNode, useState} from 'react';
+import {
+  AccordionTitle,
+  AccordionContent,
+  Accordion,
+  Icon,
+  TableRow,
+  TableHeaderCell,
+  TableHeader,
+  TableCell,
+  TableBody,
+  Table,
+  TableFooter,
+  Popup,
+  Message,
+  MessageContent,
+} from 'semantic-ui-react';
+
+import {Param, Plural, PluralTranslate, Singular, Translate} from 'indico/react/i18n';
+
+import './ParticipantListAccordion.module.scss';
+
+interface TableColumnObj {
+  id: number;
+  text: string;
+  is_picture: boolean;
+}
+
+interface TableRowObj {
+  id: number;
+  checked_in: boolean;
+  columns: TableColumnObj[];
+}
+
+interface TableObj {
+  headers: string[];
+  rows: TableRowObj[];
+  num_participants: number;
+  show_checkin: boolean;
+  title?: string;
+}
+
+interface ParticipantListAccordionProps {
+  totalParticipantCount: number;
+  published: boolean;
+  tables: TableObj[];
+}
+
+interface AccordionParticipantsItemProps {
+  table: TableObj;
+  collapsible?: boolean;
+}
+
+interface ParticipantCounterProps extends HTMLAttributes<HTMLSpanElement> {
+  totalCount: number;
+  hiddenCount: number;
+  styleName?: string;
+}
+
+function ParticipantCountHidden({count}: {count: number}) {
+  return (
+    <PluralTranslate count={count}>
+      <Singular>
+        <Param name="count" value={count} /> participant registered anonymously.
+      </Singular>
+      <Plural>
+        <Param name="count" value={count} /> participants registered anonymously.
+      </Plural>
+    </PluralTranslate>
+  );
+}
+
+function ParticipantCounter({
+  totalCount,
+  hiddenCount,
+  styleName = '',
```

Comment:
I don't think any component should ever have a `styleName` prop since `styleName` is transformed to `className` automatically at compile time.

If typescript is complaining, maybe something from that issue helps: https://github.com/gajus/babel-plugin-react-css-modules/issues/292